routing/endpointregistry.go

```diff
@@ -18,6 +19,7 @@ type Metrics interface {
 
 type entry struct {
 	detected         time.Time
+	lastSeen         time.Time
 	inflightRequests int64
```

Comment:
I was thinking about turning `inflightRequests` into `atomic.Int64`, but decided not to do this, because of `func (e *EndpointRegistry) GetMetrics(key string)` methods which returns a safe-to-read copy of entry. I expect "usual reads" to be much faster that `func (x *atomic.Int64) Load()`.

Reply:
Why do we need a copy? I think we should return pointer with safe accessor methods